### PR TITLE
Integrate better bot support into mastercomfig. (#558)

### LINKED ISSUE #558: Integrate better bot support into mastercomfig.
**Is your feature request related to a problem? Please describe.**
Bots are horribly configured in stock TF2. The bot cvars (e.g `tf_bot_notice_backstab_chance`) aren't matched up with the bot difficulty (determined by `tf_bot_difficulty`), and/or are set to bad parameters.

For instance:

* Archer Sniper and Demo bots will always fire with the same slope (set through `tf_bot_arrow_elevation_rate` and `tf_bot_ballistic_elevation_rate` respectively). The slope should vary based on difficulty.

* `tf_bot_pyro_shove_away_range` should be set to ~~0~~ 1 to 50 for Hard and Expert difficulty configurations.

* `tf_bot_health_critical_ratio` and `tf_bot_health_ok_ratio` should be raised higher by ≥0.1 to match community-defined critical/ok health ratios.

* `tf_bot_always_full_reload` should be set to 0 for easy difficulty, and 1 for the other difficulty configurations.

**Describe the solution you'd like**
Integrate better bot optimization into mastercomfig. Maybe there could a module to determine the bot difficulty, and a new command to summon a preset of bots.

To automatically fix an issue where RED bots cannot move in Mann Vs. Machine, due to the RED flag not existing, an `item_teamflag` entity that performs `ForceReset` upon itself on pickup could be created upon a custom command.

EDIT: I was also thinking there could be an addon to add improved [navigation meshes](https://developer.valvesoftware.com/wiki/Navigation_Meshes) as not all of the stock TF2 maps have navigation meshes by default. `nav_generate` can somewhat remedy this problem, but the result is often a terrible navigation mesh, making bots act poorer than with a custom made nav mesh. However the size of the addon could get pretty large (due to the sizes of nav mesh files) and override other nav meshes, so I'm leaving it as a suggestion for an addon.

**Describe alternatives you've considered**
None.

**Additional context**
Since there's not much already-existing documentation on Team Fortress 2 bots, I posted as much information as I can get on the VDC wiki.<sup>[\[1\]](https://developer.valvesoftware.com/wiki/Tf_bot)[\[2\]](https://developer.valvesoftware.com/wiki/Talk:Navigation_Meshes#Team_Fortress_2_Nav_meshes)</sup> (EDIT: Sigsegv has also posted information on [difficulty levels themselves](https://steamcommunity.com/sharedfiles/filedetails/?id=572409016)) I don't have any skill in reverse engineering, so I'm unable to test for any discrepancies in information, but judging by how bots weren't updated that much in Jungle Inferno and after, it should be close enough to current TF2.

**NOTE: When I say Bot, I do *not* mean cheating bots; I am referring to the regular bots made by Valve back in 2010.**

### PR DESCRIPTION
**Pull request by WhyIsEvery4thYearAlwaysBad**
For #558. ~~PR shouldn't be closed until it's ready to merge, unless if the maintainer rejects it.~~ PR is ready for review.

* I decided to exec the `bot_skill` module in `game.cfg`, as having it in `game_overrides.cfg` would just have made redundant console output. Check the comments in `game.cfg` for more information. 

* I ended up having to execute the `tf_bot*` cvars in a file, because every time I tried to run them all in an alias, it would cause a buffer overflow.

* Seems like `mp_highlander 1` only causes problems for bots when adding bots *after* it's been set. Adding bots before setting `mp_highlander 1` seems to cause no problems and functions as usual.

* I'm not really sure how I should model the `tf_bot*` cvars and `bot_skill` module. Maybe base them off playtime in hours, or the knowledge of a typical player?

* The current `tf_bot` cvars haven't been tested thoroughly yet as this is just the initial setup for bot support.

### CODE REVIEW

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
The PR is ready for review now.

Also it turns out that `tf_bot_keep_class_after_death` does not run in MvM, since the game will not allow bots to change classes after death.

**Comment by micwoj92** [comment]
> The PR is ready for review now.

You said that, yet you keep adding more and more changes. It only makes reviewing harder.

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
> > The PR is ready for review now.
> 
> You said that, yet you keep adding more and more changes. It only makes reviewing harder.

Right sorry. I thought I had finished up everything, but then realized there were bugs I had to patch.

Most of my future changes at this point are just to the documentation.

**Review by mastercoms — CHANGES_REQUESTED** [review]
Could you refactor this into an addon?

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
Yes I can refactor this change into an addon.

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
Finished converting the changes into an addon. @mastercoms

File: docs/customization/modules.md [hunk 1]
```diff
@@ -713,4 +713,4 @@ Default setting: **`bot_skill=normal`** (all presets).
 * **`bot_skill=easy`**: Applies easy difficulty and optimizations to bots.
 * **`bot_skill=normal`**: Applies normal difficulty and optimizations to bots.
 * **`bot_skill=hard`**: Applies hard difficulty and optimizations to bots.
-* **`bot_skill=expert`**: Applies expert difficulty and optimizations to bots.
```
Comment (micwoj92) [inline docs/customization/modules.md]: This is only confusing.

Comment (WhyIsEvery4thYearAlwaysBad) [reply docs/customization/modules.md]: Yeah. Looking at the modules documentation again, the explanations seem confusing.

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
@mastercoms All 3 checks have passed. I'm currently waiting for your approval.

**Comment by mastercoms** [comment]
There are merge conflicts

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
@mastercoms I fixed merge conflicts.

**Comment by Tiagoquix** [comment]
Thanks for your addon suggestion, but based on my review, this addon is too complex for a very small effect/gain. In the end, few people would use it due to the amount of commands and so on.

Even so, I suggest you to do not give up and upload it to GameBanana.

**Comment by WhyIsEvery4thYearAlwaysBad** [comment]
I agree; this addon feels too convoluted for mastercomfig. I have setup [a gitlab repo with the bot-support addon](https://gitlab.freedesktop.org/WhyIsEvery4thYearAlwaysBad/mastercomfig-bot-support/-/tree/develop/) if anyone needs it though.

**Pull request closed without merging**